### PR TITLE
Examples for Typescript type parameters for cy.intercept and cy.wait

### PR DESCRIPTION
**Pull request by cacieprins**
This covers changes introduced by: https://github.com/cypress-io/cypress/pull/29508

### CODE REVIEW

File: docs/api/commands/wait.mdx [hunk 1]
```diff
@@ -16,6 +16,47 @@ cy.wait(alias, options)
...
+type UserReq = {}
+type UserRes = {}
+type ActivityReq = {}
+type ActivityRes = {}
+
+cy.intercept('/users/*').as('getUsers')
+cy.intercept('/activities/*').as('getActivities')
+
+// As templated types:
+cy.wait<CustomRequest, CustomResponse>('@getUsers').then(
```
Comment (scottmcginness) [inline docs/api/commands/wait.mdx:30]: The types here will need to match the ones above (and in the callback body):
`cy.wait<UserReq, UserRes>(...`

File: docs/api/commands/wait.mdx [hunk 2]
```diff
@@ -16,6 +16,47 @@ cy.wait(alias, options)
...
+  ({ request, response }) => {
+    request.body // will be of type UserReq
+    response.body // will be of type UserRes
+  }
+)
+
+// As inferred types, with type `Interception` available in `cypress/types/net-stubbing`
+cy.wait('@getUsers').then(
+  ({ request, response }: Interception<CustomRequest, CustomResponse>) => {
```
Comment (scottmcginness) [inline docs/api/commands/wait.mdx:39]: Similarly to above, the types should match the above (and expected by the callback body):
`...Interception<UserReq, UserRes>...`

File: docs/api/commands/wait.mdx [hunk 3]
```diff
@@ -16,6 +16,47 @@ cy.wait(alias, options)
...
+    request.body // will be of type UserReq
+    response.body // will be of type UserRes
+  }
+)
+
+// When passing an array of aliases, types must me inferred:
```
Comment (scottmcginness) [inline docs/api/commands/wait.mdx:45]: Looks like a typo here :)
`me` → `be`

File: docs/api/commands/wait.mdx [hunk 4]
```diff
@@ -16,6 +16,47 @@ cy.wait(alias, options)
...
+cy.wait(['@getUsers', 'getActivities']).then(
+  (
+    interceptions: Array<
+      Interception<UserReq | ActivityReq, UserRes, ActivityRes>
```
Comment (scottmcginness) [inline docs/api/commands/wait.mdx:49]: Small typo here:
`Interception<UserReq | ActivityReq, UserRes | ActivityRes>`
(note the comma → pipe)

**Comment by cacieprins** [comment]
@scottmcginness Thank you for your feedback! I'd gone back and forth on a few different edits, and missed updating some of them. Thanks for catching them!

**Review by AtofStryker — APPROVED** [review]

**Review by mschile — APPROVED** [review]

File: docs/api/commands/intercept.mdx [hunk 5]
```diff
@@ -1601,16 +1617,16 @@ information about the request and response to the console:
...
-| [6.0.0](/guides/references/changelog#6-0-0)   | Removed `experimentalNetworkStubbing` option and made it the default behavior.                                                                                                                                                                                                                       |
-| [5.1.0](/guides/references/changelog#5-1-0)   | Added experimental `cy.route2()` command under `experimentalNetworkStubbing` option.                                                                                                                                                                                                                 |
+| Version | Changes |
+| ----------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | |
```
Comment (CoryDanielson) [inline docs/api/commands/intercept.mdx:1621]: @cacieprins It looks like you added an unnecessry ` |` to the end of this markup

Issue opened here https://github.com/cypress-io/cypress-documentation/issues/5924

**Pull request merged**